### ISSUE TITLE
[otbn,darjeeling] Loop stack push/pop at same time

### ISSUE THREAD

**Issue by rswarbrick** [lowRISC/opentitan#20792]
### Description

This causes an assertion to fail at the top of the `integrated_dev` branch. To reproduce, the following should work:
```
util/dvsim/dvsim.py hw/ip/otbn/dv/uvm/otbn_sim_cfg.hjson -i otbn_single --fixed-seed 1161969102
```

The binary that gets generated will die (correctly) because it contains the following code snippet:
```
 a90:   000010fb                loopi          1, 1
 a94:   03c010fb                loopi          1, 61
```
The second loop instruction here will try to push to the loop stack on the last instruction of the enclosing loop.

The test fails with the following assertion failure:
```
UVM_INFO @   2725176 ps: (otbn_single_vseq.sv:29) uvm_test_top.env.virtual_sequencer [uvm_test_top.env.virtual_sequencer.otbn_single_vseq] Loading OTBN binary from `/home/rjs/work/opentitan/scratch/integrated-work/otbn-sim-xcelium/0.otbn_single/latest/otbn-binaries/1161969102.elf'
xmsim: *E,ASRTST (/home/rjs/work/opentitan/scratch/integrated-work/otbn-sim-xcelium/default/src/lowrisc_ip_otbn_0.1/rtl/otbn_loop_controller.sv,317): (time 55067699 PS) Assertion tb.dut.u_otbn_core.u_otbn_controller.u_otbn_loop_controller.NoLoopStackPushAndPop has failed 
xmsim: *E,ASRTST (/home/rjs/work/opentitan/scratch/integrated-work/otbn-sim-xcelium/default/src/lowrisc_ip_otbn_0.1/rtl/otbn_loop_controller.sv,318): (time 55067699 PS) Assertion tb.dut.u_otbn_core.u_otbn_controller.u_otbn_loop_controller.NoLoopWriteIfCounterDec has failed 
UVM_ERROR @  55067699 ps: (otbn_loop_controller.sv:317) [ASSERT FAILED] NoLoopStackPushAndPop
```
This currently passes (with the same binary) on master.

**Comment by rswarbrick** [comment]
Oh! I think the bug was caused by commit f6913b422c0, which changed the assertion with this diff:
```
-  `ASSERT(NoLoopStackPushAndPop, !(loop_stack_push && loop_stack_pop))
+  `ASSERT(NoLoopStackPushAndPop, !(loop_stack_push_req && loop_stack_pop))
   `ASSERT(NoLoopWriteIfCounterDec, current_loop_counter_dec |-> !loop_stack_write)
```

@GregAC: I *think* this is a genuine bug, since the `push_i` signal to `loop_info_stack` doesn't get squashed (because there was a push request, but we don't want to honour it).

**Comment by rswarbrick** [comment]
Looking more carefully at the commit, I think we're probably ok (because there's a `loop_stack_commit` signal that should be squashing the simultaneous push+pop). So do we just need to qualify the assertion with the commit signal?

**Comment by GregAC** [comment]
@rswarbrick yes that should do it. Without commit set it doesn't matter what the `loop_stack_push_req` and `loop_stack_pop` are doing as no stack state change will occur.

That said I'm a little surprised we're seeing both the push and pop set here, I better reproduce and look and what's produce this just in case. Thanks for catching it!

**Comment by GregAC** [comment]
@rswarbrick yes that should do it. Without commit set it doesn't matter what the `loop_stack_push_req` and `loop_stack_pop` are doing as no stack state change will occur.

That said I'm a little surprised we're seeing both the push and pop set here, I better reproduce and look and what's produce this just in case. Thanks for catching it!

**Comment by GregAC** [comment]
I've taken a look at this and realised we're seeing scenarios where a simultaneous push and pop is accompanied by a commit, which isn't good. However this only happens when we're seeing a software error so OTBN will halt and the stack will be reset before the next execution anyway so ultimately doesn't matter.

Still I think worth fixing in the RTL, which is handled by this PR: https://github.com/lowRISC/opentitan/pull/22171

**Comment by etterli** [comment]
Fixed with [#22171](https://github.com/lowRISC/opentitan/pull/22171).
